mavros_extras/src/plugins/distance_sensor.cpp

```diff
@@ -55,6 +57,34 @@ class DistanceSensorItem {
 
 	void range_cb(const sensor_msgs::Range::ConstPtr &msg);
 	static Ptr create_item(DistanceSensorPlugin *owner, std::string topic_name);
+
+private:
+	std::vector<float> data;
+	bool cov_is_def = false;
+	/**
+	 * Calculate measurements variance to send to the FCU.
+	 */
+	float calculate_variance(float range) {
+		data.push_back(range);
+
+		float average, variance, sum = 0, sum_ = 0;
+
+		/*  Compute the sum of all elements */
+		for (auto i : data)
+		{
+			sum = sum + data.at(i);
```

Comment:
No! Please read about new `for`: http://www.cprogramming.com/c++11/c++11-ranged-for-loop.html